### ISSUE TITLE
Winnow Facility dropdown on VBA Facility Service to match editor's Workbench Access Section

### ISSUE THREAD

**Issue by davidmpickett** [department-of-veterans-affairs/va.gov-cms#15901]
## User Story or Problem Statement
When VBA Facility Service content type was first built, we didn't have enough sections set up under VBA to do any sort of winnowing of the Facility Drop Down list - https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11197#issuecomment-1289648669

While QAing #11334, I discovered that as a local VBA editor assigned to the Cheyenne section, I can create a VBA Facility service for VBA Facilities in different sections

![Screenshot 2023-10-30 081002](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/73b7e25f-5dc9-4e6b-8677-015acee0f401)

![screencapture-cms-uuviupwhkcoop9okwkjb5fw3j1rdyqtj-ci-cms-va-gov-node-62767-2023-10-30-08_15_00](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/e2ea0d8f-2719-4755-b161-9ba87851c3ad)

## Steps for Implementation
- [ ] Update `local_facilities_entity_reference_view` to have a new display that is unique to VBA instead of just applying to VAMCs?
  - [ ] Append the parent name to the facility in the dropdown element.
- [ ] Update `field_office` on `vba_facility_service` to use an entity reference filter by view to use the display.
- [ ] If needed, update the javascript for VBA Facility
- [ ] Update the caching on the VBA user in va_gov_user

## Acceptance Criteria
- [ ] A user with the role VBA - Content Creator can only create VBA Facility Services for Facilities that are in the same Section(s) as that User
- [ ] Cypress test(s) are created to account for this.

**Comment by mmiddaugh** [comment]
Does this account for the the Safari wrinkle documented in #12884?

**Comment by omahane** [comment]
> Does this account for the the Safari wrinkle documented in #12884?

That is a different issue, having to do with the way Safari handles styling, which differs from Chrome, Firefox or Edge.

**Comment by davidmpickett** [comment]
Update from a call with @omahane and @swirtSJW:
-  @omahane discovered that the core winnower code is not working as intended in places where it was already applied. Best theory on this is that the winnower was coded in a way that worked on a previous version of Drupal (possibly 7 or 8) and broke on a previous upgrade (not the D10 upgrade).
- In order to get the winnower to function correctly for VBA, he has to fix the core issue, which may result in also fixing other winnower defects on various other places and will add winnower function to places it wasn't previously applied (but would likely benefit from it)
- Other tickets that might be fixed by this ticket
  - #15391
  - #7538
  - #12737
  - #15906

**Comment by davidmpickett** [comment]
@jilladams @xiongjaneg 
TLDR of last comment - the scope of this ticket is likely larger than VBA and may involve cross-team code review. However, given the fact that this could close multiple outstanding defect tickets, seems like that is a good thing. It would be more effort for @omahane to try restrict his work just to VBA.

**Comment by xiongjaneg** [comment]
tagging @mmiddaugh for awareness as we had discussed having some capacity on the Drupal side as Front end catches up on VBA work so a bit of scope creep may be appropriate on this and payoff in the long run

**Comment by omahane** [comment]
@xiongjaneg @jilladams I have notified Public Websites and have solicited their feedback on this change: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15391#issuecomment-1879265840

**Comment by jilladams** [comment]
Added tags for other products that'll be updated by this change.
